packages/accounts/src/lib.rs

```diff
@@ -0,0 +1,153 @@
+//! # Soroban Smart Accounts
+//!
+//! A flexible and modular smart account framework for Soroban that enables
+//! advanced authentication and authorization patterns through composable rules,
+//! signers, and policies.
+//!
+//! ## Overview
+//!
+//! This crate provides a context-centric smart account implementation that
+//! manages the composition of authorization intents from various sources. It
+//! separates the "who" (signers), "what" (rules for specific actions), and
+//! "how" (pluggable policies) of authorization, enabling flexible combinations
+//! of authentication methods.
+//!
+//! Smart accounts can combine multiple authorization mechanisms seamlessly. For
+//! example, a wallet might require both a session policy and a passkey with a
+//! 24-hour lifespan, treating them as a single composite "key" for signing
+//! actions.
+//!
+//! ## Core Components
+//!
+//! ### Context Rules
+//!
+//! Context Rules define the scope of authorization and compose signers with
+//! policies:
+//!
+//! - Set authorization requirements for specific context types (contract calls,
+//!   deployments)
+//! - Must contain at least one signer or one policy
+//! - Support optional expiration via ledger sequence (`valid_until`)
+//! - Allow multiple rules for the same context with different configurations
+//! - Evaluated in reverse chronological order (most recent first)
+//!
+//! Supported context types:
```

Comment:
I'm still confused about these, 1-2 examples for especially `Default` and `CallContract` would be useful on when to use which. No need for elaborate examples, but smt like (i.e. blah blah)